packages/m/msgpack-cxx/xmake.lua

```diff
@@ -1,5 +1,5 @@
 package("msgpack-cxx")
-
+    set_kind("library", {headeronly = true})
```

Comment:
we should set it dynamicaly in on_load. use `package:config("header_only")`

Reply:
setting it dynamically in on_load would not work as the cmake file does not generate a .lib file if it is not linked with boost

Reply:
package:config("header_only") should be consistent with the state of set_kind("library", {headeronly = true}), otherwise there is no point in adding `header_only config`.

If the library is always headeronly, we shouldn't add add_configs("header_only")